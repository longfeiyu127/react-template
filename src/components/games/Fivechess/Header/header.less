@import '../../../../assets/styles/theme.less';
.g-fic-header{
  height: 1.8rem;
  background: url('../../../../assets/images/games/fivechess/header.jpg') no-repeat;
  background-size: 100% 100%;
  position: relative;
  .maximize{
    position: absolute;
    top: .2rem;
    right: .3rem;
    z-index: @modal-zindex;
  }
  .player {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 0.388rem;
    width: 0.388rem;
    position: absolute;
    right: 0.2rem;
    bottom: 0.2rem;
    .spark{
      position: absolute;
      height: 100%;
      width: 100%;
      background: url('../../../../assets/images/games/fivechess/spark.png') no-repeat;
      background-size: 100% 100%;
      animation: rotate 1s linear infinite;
    }
    .piece{
      height: .28rem;
      width: .28rem;
    }
    .pieceBlack{
      background: url('../../../../assets/images/games/fivechess/playerBlack.png') no-repeat;
      background-size: 100% 100%;
    }
    .pieceWhite{
      background: url('../../../../assets/images/games/fivechess/playerWhite.png') no-repeat;
      background-size: 100% 100%;
    }
  }
  @keyframes rotate {
    to {
      transform: rotate(360deg)
    }
  }
}