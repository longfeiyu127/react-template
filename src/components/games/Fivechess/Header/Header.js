import React from 'react';
import './header.less';

export default function Header(props) {
  // eslint-disable-next-line
  const { fullScreen, isFullscreen, winner, isBlack } = props;
  return (
    <header className="g-fic-header">
      <div
        onClick={fullScreen}
        className={[
          'maximize',
          'iconfont',
          isFullscreen() ? 'icon-minimize' : 'icon-maximize'
        ].join(' ')}
        name="fullScreen"
      />
      <div className="player">
        <div className={['spark', winner ? 'hide' : ''].join(' ')} />
        <div
          className={['piece', isBlack ? 'pieceBlack' : 'pieceWhite'].join(' ')}
        />
      </div>
    </header>
  );
}
