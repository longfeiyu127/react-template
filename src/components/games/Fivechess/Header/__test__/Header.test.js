import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';

Enzyme.configure({ adapter: new Adapter() });

describe('<Header />', () => {
  it('component render', () => {
    const fn = () => true;
    let header = renderer.create(<Header isFullscreen={fn} />).toJSON();
    expect(header).toMatchSnapshot();

    header = renderer.create(<Header isFullscreen={fn} winner />).toJSON();
    expect(header).toMatchSnapshot();

    header = renderer.create(<Header isFullscreen={fn} isBlack />).toJSON();
    expect(header).toMatchSnapshot();

    header = renderer
      .create(<Header isFullscreen={fn} winner isBlack />)
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  it('mount', () => {
    const fn = jest.fn();
    const gameover = mount(
      <Header fullScreen={fn} isFullscreen={fn} winner isBlack />
    );
    gameover.find("[name='fullScreen']").simulate('click');
    expect(fn.mock.calls.length).toBe(1);
  });
});
