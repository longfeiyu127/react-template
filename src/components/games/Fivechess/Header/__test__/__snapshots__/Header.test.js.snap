// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component render 1`] = `
<header
  className="g-fic-header"
>
  <div
    className="maximize iconfont icon-minimize"
    name="fullScreen"
  />
  <div
    className="player"
  >
    <div
      className="spark "
    />
    <div
      className="piece pieceWhite"
    />
  </div>
</header>
`;

exports[`<Header /> component render 2`] = `
<header
  className="g-fic-header"
>
  <div
    className="maximize iconfont icon-minimize"
    name="fullScreen"
  />
  <div
    className="player"
  >
    <div
      className="spark hide"
    />
    <div
      className="piece pieceWhite"
    />
  </div>
</header>
`;

exports[`<Header /> component render 3`] = `
<header
  className="g-fic-header"
>
  <div
    className="maximize iconfont icon-minimize"
    name="fullScreen"
  />
  <div
    className="player"
  >
    <div
      className="spark "
    />
    <div
      className="piece pieceBlack"
    />
  </div>
</header>
`;

exports[`<Header /> component render 4`] = `
<header
  className="g-fic-header"
>
  <div
    className="maximize iconfont icon-minimize"
    name="fullScreen"
  />
  <div
    className="player"
  >
    <div
      className="spark hide"
    />
    <div
      className="piece pieceBlack"
    />
  </div>
</header>
`;
